pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -459,7 +459,7 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 
 	Context("with VirtualMachineInstance spec", func() {
 		It("should accept valid machine type", func() {
-			vmi := api.NewMinimalVMI("testvmi")
+			vmi := newBaseVmi()
 			if webhooks.IsPPC64(&vmi.Spec) {
```

Comment:
out of scope:
This if-else makes no sense...

Reply:
is this better:
```
		DescribeTable("should accept valid machine type", func(arch string, expectedMachineType string) {
				vmi := newBaseVmi(libvmi.WithArchitecture(arch))
				vmi.Spec.Domain.Machine = &v1.Machine{Type: expectedMachineType}

				causes := ValidateVirtualMachineInstanceSpec(k8sfield.NewPath("fake"), &vmi.Spec, config)
				Expect(causes).To(BeEmpty())
			},
			Entry("for PPC64 architecture", "ppc64le", "pseries"),
			Entry("for ARM64 architecture", "arm64", "virt"),
			Entry("for S390X architecture", "s390x", "s390-ccw-virtio"),
			Entry("for default architecture", "", "q35"),
		)
```

Reply:
I see the first commit is not there anymore.

Reply:
i can bring it back, i just thought maybe it's too big of a change for this PR and maybe i should split the commits into separate PRs

Reply:
we can keep it separated :)

Reply:
> is this better:
> 
> ```
> 		DescribeTable("should accept valid machine type", func(arch string, expectedMachineType string) {
> 				vmi := newBaseVmi(libvmi.WithArchitecture(arch))
> 				vmi.Spec.Domain.Machine = &v1.Machine{Type: expectedMachineType}
> 
> 				causes := ValidateVirtualMachineInstanceSpec(k8sfield.NewPath("fake"), &vmi.Spec, config)
> 				Expect(causes).To(BeEmpty())
> 			},
> 			Entry("for PPC64 architecture", "ppc64le", "pseries"),
> 			Entry("for ARM64 architecture", "arm64", "virt"),
> 			Entry("for S390X architecture", "s390x", "s390-ccw-virtio"),
> 			Entry("for default architecture", "", "q35"),
> 		)
> ```

Anyway I don't think this is going to pass..
using a different arch from the runtime requires the multiarch FG, and I think that this is what we have to test

Reply:
Already addressing this in a follow-up:
```go
		DescribeTable("should accept valid machine type", func(arch string) {
			enableFeatureGate(virtconfig.MultiArchitecture)
			vmi := newBaseVmi(
				libvmi.WithArchitecture(arch),
				withMachine(&v1.Machine{Type: config.GetMachineType(arch)}),
			)

			ar, err := newAdmissionReviewForVMICreation(vmi)
			Expect(err).ToNot(HaveOccurred())

			resp := vmiCreateAdmitter.Admit(context.Background(), ar)
			Expect(resp.Allowed).To(BeTrue())
			Expect(resp.Result).To(BeNil())
		},
			Entry("when arch is ppc64le", "ppc64le"),
			Entry("when arch is arm64", "arm64"),
			Entry("when arch is s390x", "s390x"),
			Entry("when arch is amd64", "amd64"),
		)

```

Reply:
Nice